test/orange/orangeinp/detail/PolygonUtils.test.cc

```diff
@@ -149,6 +149,28 @@ TEST(PolygonUtilsTest, convex_self_intersect)
     EXPECT_FALSE(is_convex(self_int2));
 }
 
+TEST(PolygonUtilsTest, filter_colinear_points)
+{
+    auto expect_eq = [](VecReal2 const& a, VecReal2 const& b) {
+        EXPECT_EQ(a.size(), b.size());
+        for (auto i : range(a.size()))
+        {
+            EXPECT_EQ(a[i][0], b[i][0]);
+            EXPECT_EQ(a[i][1], b[i][1]);
+        }
+    };
+
+    using VecReal2 = std::vector<Real2>;
+    real_type tol = 0.01;
+
+    // Nothing colinear
+    VecReal2 corners{{0, 0}, {0, 1}, {1, 1}, {1, 0}};
+    expect_eq(corners, make_colinear_mask(make_span(corners), tol));
+
+    // Nothing colinear
+    VecReal2 corners{{0, 0}, {0, 1}, {1, 1}, {1, 0}};
+    expect_eq(corners, make_colinear_mask(make_span(corners), tol));
+}
```

Comment:
An interesting test to add would be to have a very wide tolerance and generate 1000 points on a circle and see what they filter down to.

Reply:
Yep I rewrote the code yesterday to handle this case. Will add.